src/renderer/components/dialogs/ViewGroup.tsx

```diff
@@ -110,22 +115,36 @@ export const useGroup = (chat: Type.FullChat) => {
   const [groupImage, setGroupImage] = useState(chat.profileImage)
 
   useEffect(() => {
-    ;(async () => {
-      modifyGroup(chat.id, groupName, groupImage, groupMembers)
-    })()
+    modifyGroup(chat.id, groupName, groupImage, groupMembers)
   }, [groupName, groupImage, groupMembers, chat.id])
 
   const removeGroupMember = (contactId: number) =>
     setGroupMembers(members => members.filter(mId => mId !== contactId))
-  const addGroupMember = (contactId: number) =>
-    setGroupMembers(members => [...members, contactId])
+
+  const addGroupMembers = async (newGroupMembers: number[]) => {
+    const accountId = selectedAccountId()
+
+    // We can only add verified contacts to a protected group
+    if (chat.isProtected) {
+      const membersVerified = await areAllContactsVerified(
+        accountId,
+        newGroupMembers
+      )
+      if (!membersVerified) {
+        // @TODO: Show dialogue here
```

Comment:
I'm not sure what the UX flow is for when a user tries to add an unverified contact to a verified group. I had a look at the [https://github.com/deltachat/deltachat-desktop/pull/3328](https://github.com/deltachat/deltachat-desktop/pull/3492/files#) flow. I assume we want to show the same dialogue (ProtectionBrokenDialog) then but a different QR code which allows the contact to scan it and join the group in a verified manner?

Reply:
it is just not possibe to add unverified members, they must not even be selected

instead, an alert with "scan qr" and "learn more" is shown, but none of them will add unverified contacts to selected contacts

however, if the "qr code" scan allows to be opened while the add-member-dialog is still open (android/ios allow that), the list should be updated if verification-state changes

Reply:
> instead, an alert with "scan qr" and "learn more" is shown, but none of them will add unverified contacts to selected contacts

Super, thank you! This should be implemented now.

> however, if the "qr code" scan allows to be opened while the add-member-dialog is still open (android/ios allow that), the list should be updated if verification-state changes

I'm currently looking into that, not sure if it is part of this PR though as there are some edge cases to take into consideration:

1. Automatic update of newly added email address within the "add member" dialogue
2. Automatic update of existing addresses (for this I found the `ContactsChanged` event, but doesn't seem to be working, I'm still troubleshooting)